daily-challenges/2024-09/14/thao.ha93.spec.ts

```diff
@@ -0,0 +1,48 @@
+function calculateBMI(weight: number, height: number) {
+    let BMI = 0;
+    if (weight <= 0 && height <= 0) {
+        throw new Error('Please recheck height or weight');
+    }
+    BMI = weight / (height * height);
+    console.log(`Kết quả BMI: ${BMI}`);
+
+    if (BMI < 18.5) {
+        console.log('Phân loại: "Gầy"');
+    }
+    else if (BMI >= 18.5 && BMI < 24.9) {
+        console.log('Phân loại: "Bình thường"');
+    }
+    else if (BMI >= 25 && BMI < 29.9) {
+        console.log('Phân loại: "Thừa cân"');
+    }
+    else if (BMI > 30) {
+        console.log("Béo phì");
+    }
+}
+let weight = 53, height = 1.5;
+calculateBMI(weight, height);
+
+//Playwright
+import { expect, test } from '@playwright/test';
+test('Day 14', async ({ page }) => {
+    await page.goto('https://material.playwrightvn.com/');
+
+    await page.locator('[href|="01"]').click();
+
+    let username= 'ThaoHa';
+    const textName = await page.locator('#username');
+    await textName.clear();
+    await textName.fill(username);
+
+    let email='lily@gmail.com'
+    const txtemail = await page.locator('#email');
+    await txtemail.clear();
+    await txtemail.fill(email);
+
+    await page.locator('[type="submit"]').click();
+
+    expect(await page.locator('//*[@id="userTable"]//tbody/tr[1]/td[2]').textContent()).toEqual(username);
+    expect(await page.locator('//*[@id="userTable"]//tbody/tr[1]/td[3]').textContent()).toEqual(email);
```

Comment:
```suggestion
    await expect(page.locator('//*[@id="userTable"]//tbody/tr[1]/td[2]')).toHaveText(username);
    await expect(page.locator('//*[@id="userTable"]//tbody/tr[1]/td[3]')).toHaveText(email);
```

Nên sử dụng web-first assertion để tránh bị flaky nha.